src/pages/workspaces/workspace/analysis/AppLauncher.js

```diff
@@ -27,8 +28,13 @@ const ApplicationLauncher = _.flow(
   })
 )(({
   name: workspaceName, sparkInterface, analysesData: { runtimes, refreshRuntimes },
-  application, workspace: { azureContext, workspace: { workspaceId, googleProject, bucketName } }
+  application, workspace: { azureContext, workspace: { workspaceId, googleProject, bucketName }, namespace }
 }, _ref) => {
+  useEffect(() => {
```

Comment:
?: As a fairly new React programmer, I understand `useEffect` to pop off whenever this component's DOM is rendered. Presumably that's only going to happen once for the ApplicationLauncher, but I know that `useEffect` in general is meant to replace the variety of hooks like `onMount` etc which annotate different lifecycle events. Is there a peculiarity in how `ApplicationLauncher` is written which ensures that this analytics event is only going to fire once, when the appLauncher loads?

Reply:
To respond to your question - this should only fire if any of the variables in the list change.